internal/campaigns/executor.go

```diff
@@ -210,9 +210,20 @@ func (x *executor) do(ctx context.Context, task *Task) (err error) {
 				diff = result.Commits[0].Diff
 			}
 
+			status.Cached = true
+
+			// If the cached result resulted in an empty diff, we don't need to
+			// add it to the list of specs that are displayed to the user and
+			// send to the server. Instead, we can just report that the task is
+			// complete and move on.
+			if len(diff) == 0 {
+				status.FinishedAt = time.Now()
```

Comment:
I think this line could also be moved up and deduped

Reply:
I thought about it, but kept it there because `createChangesetSpec()` really marks the end of executing the task in the normal case, and I'd like to account for that time as well.